strawman-Array.prototype.last.md

```diff
@@ -0,0 +1,102 @@
+# Array.prototype.last
+A proposal for an ECMAScript native property
+
+Code
+---
+
+    //sample code
+    Object.defineProperty(Array.prototype, 'last', {
+        configurable: true,
+        get         : function () {
+            if (this === null || typeof this.length !== "number") {
+                //verify if reference is still an array
+                throw new TypeError;
+            }
+            if (this.length < 0 || this.length > 2147483647) {
+                //checking for inclusive range of 0 to 2^31
+                //2^31 is the 2gb numeric threshold
+                throw new RangeError;
+            }
+            if (this.length === 0) {
+                //undefined would be returned anyways, but this condition
+                //specifies such as intentional behavior and not an oversight
```

Comment:
If `String.prototype.last` was a thing, you wouldn't expect `var str = 'foo'; str[-1] = 'a'; str.last` to give 'a', you'd expect 'o' - just as you'd expect `str.slice(-1)` to give 'o'. The same applies to `Array.prototype.last`.